image-inspector.go

```diff
@@ -103,11 +103,42 @@ func generateRandomName() string {
 	return fmt.Sprintf("image-inspector-%016x", n)
 }
 
+func getAuthConfigs(dockercfg, username, password_file *string) *docker.AuthConfigurations {
+	imagePullAuths := &docker.AuthConfigurations{
+		map[string]docker.AuthConfiguration{"": docker.AuthConfiguration{}}}
+	if *dockercfg != "" {
+		reader, err := os.Open(*dockercfg)
+		if err != nil {
+			log.Fatalf("Unable to open docker config file: %v\n", err)
+		}
+		if imagePullAuths, err = docker.NewAuthConfigurations(reader); err != nil {
+			log.Fatalf("Unable to parse docker config file: %v\n", err)
+		}
+		if len(imagePullAuths.Configs) == 0 {
+			log.Fatalf("No auths were found in the given dockercfg file")
+		}
+	}
+	if *username != "" {
+		token, err := ioutil.ReadFile(*password_file)
+		if err != nil {
+			log.Fatalf("Unable to read password file: %v\n", err)
+		}
+		imagePullAuths = &docker.AuthConfigurations{
+			map[string]docker.AuthConfiguration{"": docker.AuthConfiguration{Username: *username, Password: string(token)}}}
+	}
+
+	return imagePullAuths
+}
+
 func main() {
+	var err error
```

Comment:
I do need it for lines 170-175, do you want me to move it there? it makes sense to move it but I am used to declare variables at the top.. I am new to this language :)